android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMPBrowserActivity.java

```diff
@@ -27,16 +27,28 @@
 import com.tavultesoft.kmea.KMManager;
 import com.tavultesoft.kmea.KMManager.Tier;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class KMPBrowserActivity extends AppCompatActivity {
   private static final String TAG = "KMPBrowserActivity";
+  public static final String KMP_PRODUCTION_HOST = "https://keyman.com";
+  public static final String KMP_STAGING_HOST = "https://staging-keyman-com.azurewebsites.net";
+
+  // String used to query for cloud keyboard downloads
+  private static final String KMP_SEARCH_KEYBOARDS_FORMATSTR = "%s/go/android/%s/download-keyboards%s";
+  private static final String KMP_SEARCH_KEYBOARDS_LANGUAGES = "/languages/%s";
   private WebView webView;
-  private static final String KMP_PRODUCTION_HOST = "https://keyman.com";
-  private static final String KMP_STAGING_HOST = "https://staging-keyman-com.azurewebsites.net";
-  private static final String KMP_SEARCH_URL_FORMATSTR = "%s/keyboards%s?embed=android&version=%s";
-  private static final String KMP_LANGUAGE_FORMATSTR = "/languages/%s";
   private boolean isLoading = false;
   private boolean didFinishLoading = false;
 
+  // String used to parse for keyboard download link
+  private static final String KMP_INSTALL_KEYBOARDS_FORMATSTR = "^(%s|%s)(/keyboards/install/)(\\w+)(&bcp47=)?(.+)?";
```

Comment:
I don't think this pattern is correct; it won't match, for example, any of the following:

```
https://keyman.com/keyboards/install/khmer_angkor?bcp47=km     # normal pattern
https://keyman.com/keyboards/install/lao%20unicode  # legacy keyboard within download dialog
https://keyman.com/keyboards/install/khmer_angkor?_gid=112233&bcp47=lo  # other query params appearing
```

Again, we need to be careful to be using a URI parser to break the URL into components before attempting pattern matching (so we match on the path component only, and then extract the `bcp47` query param using the URI parser results (ref: https://docs.google.com/document/d/1rhgMeJlCdXCi6ohPb_CuyZd0PZMoSzMqGpv1A8cMFHY/edit#heading=h.qw7pas2adckj)

Perhaps we can add those examples to the unit test.

Reply:
The pattern in `KMPBrowserActivity` was replaced with an updated one in `KMPLink`.

I'll add these test cases to the unit test.